tests/test_robot.py

```diff
@@ -599,3 +609,21 @@ async def get_geometries(
 
             await client.close()
             Registry._SUBTYPES[Arm.SUBTYPE].create_rpc_client = old_create_client
+
+    @pytest.mark.asyncio
+    async def test_shutdown(self, service: RobotService):
+        async with ChannelFor([service]) as channel:
+
+            async def shutdown_client_mock(self):
+                return await self._client.Shutdown(ShutdownRequest())
+
+            client = await RobotClient.with_channel(channel, RobotClient.Options())
+
+            with mock.patch("viam.robot.client.RobotClient.shutdown") as shutdown_mock:
```

Comment:
Why do we need `mock.patch` here? I don't see it elsewhere in this file.

Reply:
`shutdown()` doesn't return anything (may only potentially raise an error), so there is no way to check any properties of it actually being called i.e. writing a non-trivial test. By `mock.patch`ing it we can use `assert_called_once()` to check that it is actually being called (and also mock a return value that it receives from the mock shutdown service!)

Reply:
I know I'm the one who helped with this idea, but I'm realizing now that you could've just changed the service implementation to raise an error with a message, and then in this test check to see if that specific error and specific message was raised. 

This also works.